internal/command/permission_checks.go

```diff
@@ -13,6 +13,8 @@ import (
 
 type PermissionCheck func(resourceOwner, aggregateID string) error
 
+type UserGrantPermissionCheck func(projectID, projectGrantID string) PermissionCheck
```

Comment:
Why is the permissionCheck only with projectID and projectGrantID? You have the resourceOwner, as you either have to check if the resource exist, or the dependent resources exist.

Reply:
Yes, but the resource owner is always the same. You pass it when you construct this permission check.